src/MissionManager/ComplexMissionItem.cc

```diff
@@ -396,6 +396,24 @@ void ComplexMissionItem::_intersectLinesWithPolygon(const QList<QLineF>& lineLis
     }
 }
 
+/// Adjust the line segments such that they are all going the same direction with respect to going from P1->P2
+void ComplexMissionItem::_adjustLineDirection(const QList<QLineF>& lineList, QList<QLineF>& resultLines)
+{
+    for (int i=0; i<lineList.count(); i++) {
+        const QLineF& line = lineList[i];
+        QLineF adjustedLine;
+
+        if (line.angle() > 180.0) {
```

Comment:
One more piece of information. The entry point for the survey is still not yet implemented and at this point is arbitrary based on polygon intersection. This endpoint flipping code will affect it as well. But the user themselves really needs control over it. That is NYI at this point. I'm not expecting them to set grid angle to adjust that. Or at least not at this point, since I haven't thought that much about it.